glue/viewers/scatter/qt/tests/test_data_viewer.py

```diff
@@ -914,61 +914,90 @@ def test_limits_log_widget_fullsphere(self):
     def test_apply_roi_polar(self):
         self.viewer.add_data(self.data)
         viewer_state = self.viewer.state
-        roi = RectangularROI(0, 0.5, 0, 0.5)
+        roi = RectangularROI(0.5, 1, 0.5, 1)
         viewer_state.plot_mode = 'polar'
         viewer_state.full_circle()
         assert len(self.viewer.layers) == 1
 
-        self.viewer.apply_roi(roi)
-
-        assert len(self.viewer.layers) == 2
-        assert len(self.data.subsets) == 1
-
-        assert_allclose(self.data.subsets[0].to_mask(), [1, 0, 0, 0])
-
-        state = self.data.subsets[0].subset_state
-        assert isinstance(state, RoiSubsetState)
-        assert state.pretransform
-        pretrans = state.pretransform
-        assert isinstance(pretrans, ProjectionMplTransform)
-        assert pretrans._state['projection'] == 'polar'
-        assert_allclose(pretrans._state['x_lim'], [viewer_state.x_min, viewer_state.x_max])
-        assert_allclose(pretrans._state['y_lim'], [viewer_state.y_min, viewer_state.y_max])
-        assert pretrans._state['x_scale'] == 'linear'
-        assert pretrans._state['y_scale'] == 'linear'
-        self.data.subsets[0].delete()
+        expected_mask = {
+            'degrees': [1, 1, 0, 1],
+            'radians': [0, 0, 0, 1]
+        }
 
-        viewer_state.y_log = True
-        self.viewer.apply_roi(roi)
-        state = self.data.subsets[0].subset_state
-        assert state.pretransform
-        pretrans = state.pretransform
-        assert isinstance(pretrans, ProjectionMplTransform)
-        assert pretrans._state['y_scale'] == 'log'
+        for angle_unit in ['radians', 'degrees']:
```

Comment:
```suggestion
        for angle_unit in ['radians', 'degrees']:
```
I think testing structure would benefit from instead setting this and the other test up with
```python
@pytest.mark.parametrize('angle_unit', ['radians', 'degrees'])
```

Reply:
I agree, using `parametrize` would be better. Maybe should include the expected mask as a parameter as well?

Reply:
As in `parametrize('angle_unit,expected_mask' (...)`? Had not even thought of that, but yes, that might make it even clearer.